primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -1401,6 +1407,26 @@ mod tests {
             test_gpu_e2e_template::<Bn254>().unwrap();
         }
 
+        fn test_gpu_ark_conversion_template<E: Pairing>()
```

Comment:
I like this test. But can we do more? Earlier code comment says:
> We assume that two types use the same underline repr.

In that case, we should be able to assert that the bytes are exactly the same. That way, if an incompatibility is introduced then the test will detect it immediately, no?

Reply:
This is a bit annoying but doable: [c82e230](https://github.com/EspressoSystems/jellyfish/pull/516/commits/c82e230c7294f8218df2da73304b8bfe6976261c)

Reply:
Thank you! I don't fully understand this new test but here's what I see:
1. make a slice of random new ark scalars in `scalars`
2. convert them into a IC scalars slice `ic_scalars` via `from_ark`.
3. copy the slice into a new GPU slice called `d_scalars`.
4. transform the GPU slice to Montgomery form on the GPU
5. copy the GPU scalars out of GPU and back into `ic_scalars`
6. these `ic_scalars` (in Montgomery form) should be byte-for-byte identical to the original `scalars`. Check this via `align_to`

Not sure why you need to copy to/from GPU but AFIACT this test does what it's supposed to.